deploy/examples/import-external-cluster.sh

```diff
@@ -76,87 +76,124 @@ function importClusterID() {
+  if [ -s a.txt ]; then
+    kubectl -n "$NAMESPACE" \
+      create \
+      secret \
+      generic \
+      --type="kubernetes.io/rook" \
+      "rook-""$CSI_RBD_PROVISIONER_SECRET_NAME" \
+      --from-literal=userID="$CSI_RBD_PROVISIONER_SECRET_NAME" \
+      --from-literal=userKey="$CSI_RBD_PROVISIONER_SECRET"
+  else
+    echo "secret $secret already exists"
+  fi
 }
 
 function importCsiCephFSNodeSecret() {
-  kubectl -n "$NAMESPACE" \
-    create \
-    secret \
-    generic \
-    --type="kubernetes.io/rook" \
-    "rook-""$CSI_CEPHFS_NODE_SECRET_NAME" \
-    --from-literal=adminID="$CSI_CEPHFS_NODE_SECRET_NAME" \
-    --from-literal=adminKey="$CSI_CEPHFS_NODE_SECRET"
+  secret=$(kubectl -n "$NAMESPACE" get secret "rook-""$CSI_CEPHFS_NODE_SECRET_NAME" 2>a.txt)
+  if [ -s a.txt ]; then
+    kubectl -n "$NAMESPACE" \
+      create \
+      secret \
+      generic \
+      --type="kubernetes.io/rook" \
+      "rook-""$CSI_CEPHFS_NODE_SECRET_NAME" \
+      --from-literal=adminID="$CSI_CEPHFS_NODE_SECRET_NAME" \
+      --from-literal=adminKey="$CSI_CEPHFS_NODE_SECRET"
+  else
+    echo "secret $secret already exists"
+  fi
 }
 
 function importCsiCephFSProvisionerSecret() {
-  kubectl -n "$NAMESPACE" \
-    create \
-    secret \
-    generic \
-    --type="kubernetes.io/rook" \
-    "rook-""$CSI_CEPHFS_PROVISIONER_SECRET_NAME" \
-    --from-literal=adminID="$CSI_CEPHFS_PROVISIONER_SECRET_NAME" \
-    --from-literal=adminKey="$CSI_CEPHFS_PROVISIONER_SECRET"
+  secret=$(kubectl -n "$NAMESPACE" get secret "rook-""$CSI_CEPHFS_PROVISIONER_SECRET_NAME" 2>a.txt)
+  if [ -s a.txt ]; then
+    kubectl -n "$NAMESPACE" \
+      create \
+      secret \
+      generic \
+      --type="kubernetes.io/rook" \
+      "rook-""$CSI_CEPHFS_PROVISIONER_SECRET_NAME" \
+      --from-literal=adminID="$CSI_CEPHFS_PROVISIONER_SECRET_NAME" \
+      --from-literal=adminKey="$CSI_CEPHFS_PROVISIONER_SECRET"
+  else
+    echo "secret $secret already exists"
+  fi
 }
 
 function importRGWAdminOpsUser() {
-  kubectl -n "$NAMESPACE" \
-    create \
-    secret \
-    generic \
-    --type="kubernetes.io/rook" \
-    "$RGW_ADMIN_OPS_USER_SECRET_NAME" \
-    --from-literal=accessKey="$RGW_ADMIN_OPS_USER_ACCESS_KEY" \
-    --from-literal=secretKey="$RGW_ADMIN_OPS_USER_SECRET_KEY"
+  secret=$(kubectl -n "$NAMESPACE" get secret "$RGW_ADMIN_OPS_USER_SECRET_NAME" 2>a.txt)
+  if [ -s a.txt ]; then
+    kubectl -n "$NAMESPACE" \
+      create \
+      secret \
+      generic \
+      --type="kubernetes.io/rook" \
+      "$RGW_ADMIN_OPS_USER_SECRET_NAME" \
+      --from-literal=accessKey="$RGW_ADMIN_OPS_USER_ACCESS_KEY" \
+      --from-literal=secretKey="$RGW_ADMIN_OPS_USER_SECRET_KEY"
+  else
+    echo "secret $secret already exists"
+  fi
 }
 
 function createECRBDStorageClass() {
-  cat <<eof | kubectl create -f -
+  storageclass=$(kubectl -n "$NAMESPACE" get storageclass $RBD_STORAGE_CLASS_NAME 2>a.txt)
```

Comment:
Is a.txt deleted? Seems like it may affect multiple different functions if it is not deleted.

Reply:
Yes